src/QsCompiler/CompilerOptimizations/StatementRemoving.fs

```diff
@@ -63,6 +90,12 @@ type internal StatementRemover() =
                 |> Seq.map wrapStmt
             QsScope.New (statements, parentSymbols)
 
+        override __.onLocation _ = Null
```

Comment:
Why is the logic for stripping range information embedded into this particular transformation? What in the code relies on there not being any range information?

Reply:
The PureCircuitAPI file relies on equality comparisons between expressions to determine whether two expressions should be replaced with the same symbol or with different symbols. Range information messes up these equality comparisons.

Reply:
Ok, I agree that you probably want to strip range info, but it would make sense to do that separately form this transformation step e.g. before handing it off to the pure circuit API.

Reply:
Fixed in latest commit